src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/Utils/GraphActionSymbolUtils.cs

```diff
@@ -0,0 +1,269 @@
+					yield return (field, fieldType);
+				}
+			}
+		}
+		#endregion
+
+		#region Action Handlers
+		/// <summary>
+		/// Get the action handlers method symbols and syntax nodes from the graph. 
+		/// The <paramref name="actionsByName"/> must have <see cref="StringComparer.OrdinalIgnoreCase"/> comparer.
+		/// </summary>
+		/// <param name="graph">The graph to act on</param>
+		/// <param name="actionsByName">The actions of the graph dictionary with <see cref="StringComparer.OrdinalIgnoreCase"/> comparer</param>
+		/// <param name="pxContext">Context</param>
+		/// <param name="cancellation">Cancellation token</param>
+		/// <param name="inheritance">If true includes action handlers from the graph inheritance chain</param>
+		/// <returns></returns>
+		public static ActionHandlersOverridableCollection GetActionHandlersFromGraph(this ITypeSymbol graph, IDictionary<string, ActionInfo> actionsByName,
+																					 PXContext pxContext, CancellationToken cancellation, 
+																					 bool inheritance = true)
+		{
+			graph.ThrowOnNull(nameof(graph));
+			actionsByName.ThrowOnNull(nameof(actionsByName));
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			if (!graph.IsPXGraph(pxContext))
+			{
+				return Enumerable.Empty<GraphOverridableItem<(MethodDeclarationSyntax, IMethodSymbol)>>();
+			}
+
+			var actionHandlersByName = new GraphOverridableItemsCollection<(MethodDeclarationSyntax, IMethodSymbol)>();
+
+			GetActionHandlersFromGraphImpl(graph, actionsByName, pxContext, cancellation, inheritance)
+				.ForEach(handler => actionHandlersByName.Add(handler.Symbol.Name, handler));
+
+			return actionHandlersByName.Items;
+		}
+
+		/// <summary>
+		/// Get the action handlers symbols and syntax nodes from the graph extension.
+		/// The <paramref name="actionsByName"/> must have <see cref="StringComparer.OrdinalIgnoreCase"/> comparer.
+		/// </summary>
+		/// <param name="graphExtension">The graph extension to act on</param>
+		/// <param name="actionsByName">The actions of the graph dictionary with <see cref="StringComparer.OrdinalIgnoreCase"/> comparer</param>
+		/// <param name="pxContext">Context</param>
+		/// <param name="cancellation">Cancellation token</param>
+		/// <returns></returns>
+		public static ActionHandlersOverridableCollection GetActionHandlersFromGraphExtensionAndBaseGraph(this ITypeSymbol graphExtension,
+																				IDictionary<string, ActionInfo> actionsByName, PXContext pxContext,
+																				CancellationToken cancellation)
+		{
+			graphExtension.ThrowOnNull(nameof(graphExtension));
+			actionsByName.ThrowOnNull(nameof(actionsByName));
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			return GetActionInfoFromGraphExtension<(MethodDeclarationSyntax, IMethodSymbol)>(
+				graphExtension, pxContext, AddHandlersFromGraph, AddHandlersFromGraphExtension);
+
+			//--------------------------------------------------------Local Functions----------------------------------------------------------
```

Comment:
I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

Reply:
ok